src/Core/Components/Spacing/SpacingGenerator.cs

```diff
@@ -0,0 +1,214 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/*
+ *  To generate the `Spacing.razor.css` utilities, run the following code in a C# Console Application,
+ *  and copy the output to the `Spacing.razor.css` file.
+ *  
+ *    System.Console.WriteLine(SpacingGenerator.Script);
+ *    
+ *  Example, using https://dotnetfiddle.net/
+ *  
+ * We ran several tests to determine the number of elements (e.g. `.ml-?`) to generate, depending on requirements 
+ * and the size of the CSS file generated. We determined that 8 elements is a good compromise
+ * Spacing can be up to 32px (positive or negative).
+ *   
+ *    Count = 01   =>   Max spacing size:  4px - File size:  29 kb.
+ *    Count = 02   =>   Max spacing size:  8px - File size:  43 kb.
+ *    Count = 03   =>   Max spacing size: 12px - File size:  57 kb.
+ *    Count = 04   =>   Max spacing size: 16px - File size:  71 kb.
+ *    Count = 05   =>   Max spacing size: 20px - File size:  85 kb.
+ *    Count = 06   =>   Max spacing size: 24px - File size: 100 kb.
+ *    Count = 07   =>   Max spacing size: 28px - File size: 114 kb.
+ * => Count = 08   =>   Max spacing size: 32px - File size: 129 kb. <==
+ *    Count = 09   =>   Max spacing size: 36px - File size: 144 kb.
+ *    Count = 10   =>   Max spacing size: 40px - File size: 159 kb.
+ *    Count = 11   =>   Max spacing size: 44px - File size: 175 kb.
+ *    Count = 12   =>   Max spacing size: 48px - File size: 191 kb.
+ *    Count = 13   =>   Max spacing size: 52px - File size: 207 kb.
+ *    Count = 14   =>   Max spacing size: 56px - File size: 223 kb.
+ *    Count = 15   =>   Max spacing size: 60px - File size: 239 kb.
+ *    Count = 16   =>   Max spacing size: 64px - File size: 256 kb.
+ *    Count = 17   =>   Max spacing size: 68px - File size: 273 kb.
+ *    Count = 18   =>   Max spacing size: 72px - File size: 290 kb.
+ *    Count = 19   =>   Max spacing size: 76px - File size: 307 kb.
+ *    Count = 20   =>   Max spacing size: 80px - File size: 325 kb.
+ *    Count = 21   =>   Max spacing size: 84px - File size: 342 kb.
+ *    Count = 22   =>   Max spacing size: 88px - File size: 360 kb.
+ *    Count = 23   =>   Max spacing size: 92px - File size: 378 kb.
+ *    Count = 24   =>   Max spacing size: 96px - File size: 397 kb.
+ *    
+ *    for (int i = 1; i < 25; i++)
+ *    {
+ *        var script = SpacingGenerator.GenerateScript(i);
+ *        System.Console.WriteLine($"   Count = {i:00}   =>   Max spacing size: {i*4}px - File size:  {script.Length / 1024} kb.");
+ *    }
+ */
```

Comment:
/*
 * We ran several tests to determine the number of elements (e.g. `.ml-?`) to generate, depending on requirements 
 * and the size of the CSS file generated. We determined that 8 elements, where Spacing can then be set up to 32px 
 * (either positive or negative), gives a good compromise between options and size of the CSS file.
 * 
 *  To find what file size corresponds to what count, the following code can be run in a C# Console Application. 
 *  
 *  System.Console.WriteLine(SpacingGenerator.Script);
 *   
 *    for (int i = 1; i < 25; i++)
 *    {
 *        var script = SpacingGenerator.GenerateScript(i);
 *        System.Console.WriteLine($"   Count = {i:00}   =>   Max spacing size: {i*4}px - File size:  {script.Length / 1024} kb.");
 *    }
 */